src/main/java/com/google/step/snippet/servlets/UserSettingServlet.java

```diff
@@ -24,11 +32,71 @@
 /** Servlet that renders user setting page */
 @WebServlet("/user")
 public class UserSettingServlet extends HttpServlet {
+  private static final String WEBSITE_PARAMETER = "website";
+  private static final String LANGUAGE_PARAMETER = "language";
+  private static final String ID_PARAMETER = "id";
+  private static final String EMAIL_PARAMETER = "email";
+  private static final String USER_CLASS_PARAMETER = "UserInfo";
+  private static final String USER_PARAMETER = "user";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    // Forward the request to the template (which is a servlet itself).
+    UserService userService = UserServiceFactory.getUserService();
+    // In case unauthenticated user access setting page through URL.
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/");
+      return;
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity userEntity = getUserEntity(datastore, userService);
+    User user =
+        new User(
+            (String) userEntity.getProperty(ID_PARAMETER),
+            (String) userEntity.getProperty(EMAIL_PARAMETER),
+            (String) userEntity.getProperty(WEBSITE_PARAMETER),
+            (String) userEntity.getProperty(LANGUAGE_PARAMETER));
+    request.setAttribute(USER_PARAMETER, user);
     request.getRequestDispatcher("WEB-INF/templates/user_dashboard.jsp").forward(request, response);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/");
+      return;
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String website = request.getParameter(WEBSITE_PARAMETER);
+    String language = request.getParameter(LANGUAGE_PARAMETER);
+    Entity userEntity = getUserEntity(datastore, userService);
+    if (website.equals("GeeksForGeeks")
+        || website.equals("StackOverflow")
+        || website.equals("W3Schools")) {
+      userEntity.setProperty(WEBSITE_PARAMETER, website);
+    }
```

Comment:
Nit: I'd declare a constant set at the top of the class and then check if the website is in the set.

```
private static final Set<String> VALID_WEBSITES =
    Collections.unmodifiableSet(
        new HashSet<>(Arrays.asList("GeeksForGeeks", "StackOverflow", "W3Schools")));

...

if (VALID_WEBSITES.contains(website)) ...
```